plugin/pkg/scheduler/factory/factory.go

```diff
@@ -278,6 +408,25 @@ func (c *ConfigFactory) deletePodFromCache(obj interface{}) {
 	if err := c.schedulerCache.RemovePod(pod); err != nil {
 		glog.Errorf("scheduler cache RemovePod failed: %v", err)
 	}
+
+	c.invalidateCachedPredicatesOnDeletePod(pod)
+}
+
+func (c *ConfigFactory) invalidateCachedPredicatesOnDeletePod(pod *v1.Pod) {
+	// part of this case is the same as pod add.
+	c.equivalencePodCache.InvalidateCachedPredicateItemForPodAdd(pod, pod.Spec.NodeName)
+	// MatchInterPodAffinity need to be reconsidered for this node, as well as all nodes in its same failure domain.
+	// TODO(resouer) can we just do this for nodes in the same failure domain
+	c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(sets.NewString("MatchInterPodAffinity"))
+
+	invalidPredicates := sets.NewString()
+	// if this pod have these PV, cached result of disk conflict will become invalid.
+	for _, volume := range pod.Spec.Volumes {
+		if volume.GCEPersistentDisk != nil || volume.AWSElasticBlockStore != nil || volume.RBD != nil || volume.ISCSI != nil {
+			invalidPredicates.Insert("NoDiskConflict")
```

Comment:
Why not invalidate it here instead of creating this "invalidPredicates" variable?